enhancements/custom-alert-configuration.md

```diff
@@ -0,0 +1,439 @@
+
+To support this, we suggest a following CustomResourceDefinition:
+
+```go
+
+//
+// AlertExclusionConfig describes the alert exclusion configuration
+//
+// Compatibility level 4: No compatibility is provided, the API can change at any point for any reason. These capabilities should not be used by applications needing long term support.
+// +openshift:compatibility-gen:level=4
+// +genclient
+// +k8s:openapi-gen=true
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type AlertExclusionConfig struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec AlertExclusionConfigSpec `json:"spec"`
+}
+
+// AlertExclusionConfigSpec describes the alert exclusion configuration
+type AlertExclusionConfigSpec struct {
+    // subjects to exclude from the alert
+    //
+    // +optional
+    Subjects map[string]string `json:"subjects,omitempty"
+}
+
+```
+
+The owner of the alert would create the AlertExclusionConfig resource and
+manage the subjects. In the case of KubeVirt, this would include determining
+if and in what namespace KubeVirt is installed.
+
+```yaml
+kind: AlertExclusionConfig
+apiVersion: config.openshift.io/v1
+metadata:
+  name: pod-disruption-budget-at-limit
+spec:
+  subjects:
+    'system:serviceaccount:kubevirt:kubevirt-controller': 'enabled'
+```
+
+Then we would introduce a validating admission policy that would allow changes
+to the `alerts.openshift.io/PodDisruptionBudgetAtLimit` label only for these
+subjects.
+
+```yaml
+apiVersion: admissionregistration.k8s.io/v1alpha1
+kind: ValidatingAdmissionPolicy
+metadata:
+  name: poddisruptionbudget-at-limit
+spec:
+  failurePolicy: Ignore
+  paramKind:
+    apiVersion: config.openshift.io/v1
+    kind: AlertExclusionConfig
+  matchConstraints:
+    resourceRules:
+    - apiGroups:   ["policy"]
+      apiVersions: ["v1"]
+      operations:  ["CREATE", "UPDATE"]
+      resources:   ["poddisruptionbudgets"]
+    objectSelector:
+      matchExpressions:
+        - key: "alerts.openshift.io/PodDisruptionBudgetAtLimit"
+          operator: Exists
+  validations:
+    - expression: "(has(params.spec.subjects) && request.userInfo.username in params.spec.subjects &&  params.spec.subjects[request.userInfo.username] == 'enabled') ? true : !((has(object.metadata.labels) && 'alerts.openshift.io/PodDisruptionBudgetAtLimit' in object.metadata.labels) || (oldObject != null && has(oldObject.metadata.labels) && 'alerts.openshift.io/PodDisruptionBudgetAtLimit' in oldObject.metadata.labels))"
+      message: "Setting alerts.openshift.io/PodDisruptionBudgetAtLimit label is not allowed."
+      reason: Forbidden
+```
+
+And bind the parameters with:
+
+```yaml
+apiVersion: admissionregistration.k8s.io/v1alpha1
+kind: ValidatingAdmissionPolicyBinding
+metadata:
+  name: poddisruptionbudget-at-limit
+spec:
+  policyName: poddisruptionbudget-at-limit
+  paramRef:
+    name: pod-disruption-budget-at-limit-alert-opt-out
```

Comment:
I'm not all that familiar with ValidatingAdmissionPolicyBinding, but I'd have expected `paramRef.name` to match the AlertExclusionConfig `name`, and up above you currently have:

```yaml
kind: AlertExclusionConfig
apiVersion: config.openshift.io/v1
metadata:
  name: pod-disruption-budget-at-limit
...
```

not `pod-disruption-budget-at-limit-alert-opt-out`.

Reply:
thanks for noticing that, forgot to update this place